File: front/src/app/core/services/comment.service.ts
import { Service, inject } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../../environments/environment';
import { CommentResponse, CreateCommentRequest } from '../models';

@Service()
export class CommentService {
  private readonly http = inject(HttpClient);
  private readonly apiUrl = environment.apiUrl;

  getByPostId(postId: number): Observable<CommentResponse[]> {
    return this.http.get<CommentResponse[]>(`${this.apiUrl}/posts/${postId}/comments`);
  }

  create(postId: number, request: CreateCommentRequest): Observable<CommentResponse> {
    return this.http.post<CommentResponse>(`${this.apiUrl}/posts/${postId}/comments`, request);
  }
}

File: back/src/main/java/com/openclassrooms/mddapi/service/CommentService.java
package com.openclassrooms.mddapi.service;

import java.util.List;

import com.openclassrooms.mddapi.dto.CommentResponse;
import com.openclassrooms.mddapi.dto.CreateCommentRequest;
import com.openclassrooms.mddapi.exception.PostNotFoundException;

/**
 * Comments attached to posts.
 */
public interface CommentService {

    /**
     * @param postId the post to list comments for
     * @return the post's comments, oldest first
     * @throws PostNotFoundException if no post matches {@code postId}
     */
    List<CommentResponse> findByPostId(Long postId);

    /**
     * Adds a comment to a post. A user may comment on any post, even on a
     * theme they are no longer subscribed to.
     *
     * @param userId  the author's id
     * @param postId  the post being commented on
     * @param request the comment content
     * @return the created comment
     * @throws PostNotFoundException if no post matches {@code postId}
     */
    CommentResponse create(Long userId, Long postId, CreateCommentRequest request);
}
